### PR TITLE
refactor and add test

### PR DESCRIPTION
**Pull request by hirosassa**
SSIA

@kitagry Could you review this changes?

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -36,43 +36,44 @@ fn main() -> ExitCode {
...
+    });
+    let errors = targets
+        .map(|f| -> (String, Option<Diagnostic>) {
+            let key = f.into_path();
+            (
+                key.to_str().unwrap_or("").to_string(),
+                fs::File::open(key)
+                    .map(|ref mut file| analyse_sql(file))
+                    .expect("failed to open file"),
+            )
```
Comment (hirosassa) [inline src/main.rs:53]: divide long method chain into two parts.

File: src/main.rs [hunk 2]
```diff
@@ -36,43 +36,44 @@ fn main() -> ExitCode {
...
 }
 
+fn is_sql(entry: &DirEntry) -> bool {
+    entry
+        .file_name()
+        .to_str()
+        .map(|s| s.ends_with(".sql"))
+        .unwrap_or(false)
+}
+
```
Comment (hirosassa) [inline src/main.rs:76]: This code borrowed from below

https://docs.rs/walkdir/latest/walkdir/#example-skip-hidden-files-and-directories-on-unix

**Review by kitagry — APPROVED** [review]
LGTM!

**Comment by hirosassa** [comment]
Thanks!!

**Pull request merged**